api/v1alpha1/packagebundle.go

```diff
@@ -45,11 +46,27 @@ func (config *PackageBundle) FindSource(pkgName, pkgVersion string) (retSource P
 	return retSource, fmt.Errorf("package not found in bundle (%s): %s @ %s", config.ObjectMeta.Name, pkgName, pkgVersion)
 }
 
+func (config *PackageBundle) IsNewer(pkgBundle *PackageBundle) bool {
+	major, minor, build := config.GetMajorMinorBuild()
+	pkgMajor, pkgMinor, pkgBuild := pkgBundle.GetMajorMinorBuild()
+	return major < pkgMajor || minor < pkgMinor || build < pkgBuild
+}
+
+func (config *PackageBundle) GetMajorMinorBuild() (major int, minor int, build int) {
+	s := strings.Split(config.Name, "-")
+	s = append(s, "", "", "")
+	s[0] = strings.TrimPrefix(s[0], "v")
+	major, _ = strconv.Atoi(s[0])
```

Comment:
The reason I don't look at the errors is because if there is an error, they evaluate to zero which is the lowest. The name "funky" evaluates to (0, 0, 0).  Bubbling that error up would be complicated in a sort for example.

I think the best way would be add the results of this to the bundle Status and report it there if we see a zero in any member of the tuple

Reply:
Discussed I think

Reply:
> The reason I don't look at the errors is because if there is an error, they evaluate to zero which is the lowest. The name "funky" evaluates to (0, 0, 0). Bubbling that error up would be complicated in a sort for example.

Nah, it's not bad. If there's an error, you just say that it's less than---which is kinda what you're hoping to do by assuming it evaluates to 0, but you're assuming when you could be 100% sure.
 
> I think the best way would be add the results of this to the bundle Status and report it there if we see a zero in any member of the tuple

Hmm... In general, assuming anything about the returned value in the presence of a non-nil error isn't recommended. It's generally undefined. In this particular case, it is defined. `Atoi` is a helper for `ParseInt`, and `ParseInt` docs say this (emphasis mine):
>  if the value corresponding to s cannot be represented by a signed integer of the given size, err.Err = ErrRange and ***the returned value is the maximum magnitude integer of the appropriate bitSize and sign***.

...but the chance that someone's version results in the returned value being `math.MaxInt` seems pretty low. :D 

So I can sign off on this, but I personally would still rather see the error values checked and possibly logged.